src/state/wallets/walletsStore.ts

```diff
@@ -93,21 +93,29 @@ interface WalletsState {
 
 export const useWalletsStore = createRainbowStore<WalletsState>(
   (set, get) => ({
-    getIsDamaged: () => !!get().selected?.damaged,
+    getIsDamagedWallet: () => !!get().selected?.damaged,
     getIsReadOnlyWallet: () => get().selected?.type === WalletTypes.readOnly,
     getIsHardwareWallet: () => !!get().selected?.deviceId,
 
     selected: null,
     setSelectedWallet(wallet, address) {
       setSelectedWalletInKeychain(wallet);
-      set({
-        selected: {
-          ...wallet,
-        },
-      });
+
+      // ensure not memoized
```

Comment:
this is a performance improvement chatted with christian about, avoiding double-sets